tests/conftest.py

```diff
@@ -120,6 +116,94 @@ def email_generator(chars=string.ascii_letters + string.digits + '.' + '-'):
     return ''.join(random.choice(chars) for _ in range(size)) + '@tieto.com'
 
 
+@pytest.fixture(scope='module')
+def mock_ldap(db, session):
+    # this class mimics ldap object
+    class MockLdap():
+        # those properties correspond to use cases
+        # for example user not present in ldap try to log in
+        def __init__(self, user, non_user, user_not_wroc, admin):
+            self.user = user
+            self.non_user = non_user
+            self.user_not_wroc = user_not_wroc
+            self.admin = admin
+
+        # check if credentials are valid
+        def bind_user(self, user_name, passwd):
+            users = [self.user, self.admin, self.user_not_wroc]
+            check_login = [
+                user_name == usr['user_name'][0].decode() and
+                passwd == usr['passwd'][0].decode()
+                for usr in users
+            ]
+            if any(check_login):
+                return True
+            else:
+                return None
+
+        # return employee data
+        def get_object_details(self, user):
+            users = [self.user, self.admin, self.user_not_wroc]
+            for usr in users:
+                if user == usr['user_name'][0].decode():
+                    return usr
+
+    # create fake user data
+    def create_user(wroclaw_usr=True, non_user=False, admin=False):
+        login = g.person.identifier(mask='@@@@@@@@').lower()
+        email = g.person.email(['@tieto.com'])
+        passwd = g.person.password(length=12)
+        name = g.person.name()
+        surname = g.person.last_name()
+        identifier = g.person.identifier(mask='#####')
+
+        if wroclaw_usr:
+            location = 'Wroclaw'
+        else:
+            while True:
+                location = g.address.city()
+                if location != 'Wroclaw':
```

Comment:
Then this if is unnecessary. Now, during the loop, condition you will be checked once not twice